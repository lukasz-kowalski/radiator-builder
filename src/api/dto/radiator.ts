interface RawDefaultProperties {
  cost_from: number;
  id: string;
  name: string;
}

interface RawLength extends RawDefaultProperties {
  h1_value: string;
  h2_value: string;
  output_dt22_5: string;
  output_dt30: string;
  output_dt40: string;
  output_dt50: string;
  oversized: "yes" | "no";
  pieces: number;
  pipe_centres: number;
  price: number;
  radiator_length: number;
  sections: number;
}

interface RawHeight extends RawDefaultProperties {
  height: number;
  label: string;
  sku: string;
  lengths: RawLength[];
}

interface RawDepth extends RawDefaultProperties {
  columns: number;
  depth: number;
  heights: RawHeight[];
}

export interface RawRadiator extends RawDefaultProperties {
  description: string;
  models: number;
  depths: RawDepth[];
}

export interface RadiatorFamily {
  id: string;
  name: string;
}

export interface Radiator {
  radiator_cost_from: number;
  radiator_id: string;
  radiator_name: string;
  radiator_description: string;
  radiator_models: number;
  length_cost_from: number;
  length_id: string;
  length_name: string;
  h1_value: string;
  h2_value: string;
  output_dt22_5: string;
  output_dt30: string;
  output_dt40: string;
  output_dt50: string;
  oversized: "yes" | "no";
  pieces: number;
  pipe_centres: number;
  price: number;
  radiator_length: number;
  sections: number;
  height_cost_from: number;
  height_id: string;
  height_name: string;
  height: number;
  label: string;
  sku: string;
  depth_cost_from: number;
  depth_id: string;
  depth_name: string;
  columns: number;
  depth: number;
}
