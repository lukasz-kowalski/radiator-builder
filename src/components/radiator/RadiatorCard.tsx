import { RadiatorItem } from "@/api/dto/radiator";

interface Props {
  radiator: RadiatorItem;
}

export default function RadiatorCard({ radiator }: Props) {
  return <div>Test</div>;
}
