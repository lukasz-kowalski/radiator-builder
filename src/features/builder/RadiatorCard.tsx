import Image from "next/image";

import { Radiator } from "@/api/dto/radiator";

interface Props {
  radiator: Radiator;
}

export default function RadiatorCard({ radiator }: Props) {
  return (
    <div
      className="bg-white border-2 p-4 shadow-md"
      data-testid="radiator-card"
    >
      <div className="relative w-[80px] h-[110px] flex justify-self-center mb-2">
        <Image
          src={`/images/emmeline-i-465mm_${radiator.sections}-sections-preview.jpg`}
          width={80}
          height={110}
          alt={`${radiator.label}, ${radiator.radiator_length} mm, ${radiator.sections} sections`}
          loading="eager"
        />
      </div>
      <div>
        <p className="font-bold">
          {radiator.label}, {radiator.length_name}
        </p>
        <p className="text-sm">Length: {radiator.radiator_length} mm</p>
        <p className="text-sm">Price from: Â£{radiator.length_cost_from}</p>
      </div>
    </div>
  );
}
