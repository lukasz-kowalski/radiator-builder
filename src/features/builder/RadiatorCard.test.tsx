import { render, screen } from "@testing-library/react";

import RadiatorCard from "@/features/builder/RadiatorCard";
import { Radiator } from "@/api/dto/radiator";

describe("RadiatorCard", () => {
  const radiator = {
    label: "Emmeline",
    length_name: "600mm",
    radiator_length: 600,
    sections: 6,
    length_cost_from: 50,
  } as Radiator;

  it("should render radiator information correctly", () => {
    render(<RadiatorCard radiator={radiator} />);

    expect(screen.getByText("Emmeline, 600mm")).toBeInTheDocument();
    expect(screen.getByText("Length: 600 mm")).toBeInTheDocument();
    expect(screen.getByText("Price from: Â£50")).toBeInTheDocument();
  });

  it("should match the snapshot", () => {
    const { asFragment } = render(<RadiatorCard radiator={radiator} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
