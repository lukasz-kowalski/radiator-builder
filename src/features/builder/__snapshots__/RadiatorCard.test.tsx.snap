// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadiatorCard should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white border-2 p-4 shadow-md"
  >
    <div
      class="relative w-[80px] h-[110px] flex justify-self-center mb-2"
    >
      <img
        alt="Emmeline, 600 mm, 6 sections"
        data-nimg="1"
        decoding="async"
        height="110"
        loading="eager"
        src="/_next/image?url=%2Fimages%2Femmeline-i-465mm_6-sections-preview.jpg&w=256&q=75"
        srcset="/_next/image?url=%2Fimages%2Femmeline-i-465mm_6-sections-preview.jpg&w=96&q=75 1x, /_next/image?url=%2Fimages%2Femmeline-i-465mm_6-sections-preview.jpg&w=256&q=75 2x"
        style="color: transparent;"
        width="80"
      />
    </div>
    <div>
      <p
        class="font-bold"
      >
        Emmeline, 600mm
      </p>
      <p
        class="text-sm"
      >
        Length: 600 mm
      </p>
      <p
        class="text-sm"
      >
        Price from: Â£50
      </p>
    </div>
  </div>
</DocumentFragment>
`;
